### ISSUE TITLE
Suport Prisma 4.0.0, Error running version nexus: 1.3.0 Error: NEXUS__UNKNOWN__TYPE was already defined and imported as a type

### ISSUE THREAD

**Issue by reaink** [graphql-nexus/nexus#1119]
running nexus v1.3.0, prisma v4.0.0 Error：
Error: NEXUS__UNKNOWN__TYPE was already defined and imported as a type, check the docs for extending types

**Comment by Fatezhang** [comment]
Hey @reaink , can I please ask what's your current prisma version?

**Comment by reaink** [comment]
current version
```
    "prisma": "3.14.0",
    "nexus": "1.3.0",
```

update to prisma@v4.x throw error ` NEXUS__UNKNOWN__TYPE`

**Comment by kevscript** [comment]
I'm having the same error. Downgrading  to v3.15.2 didn't fix it. Its still working on a previous project with this prisma version tho.
@reaink any updates on your side?

**Comment by reaink** [comment]
@kevscript Yes, I hope Prisma 4.x can be supported, currently I'm still using prisma 3.x in current nexus version

**Comment by kostovaljubica7** [comment]
I'am having the same issue upgrading 
 3.15.2 -> 4.21
 `  "nexus": "1.3.0",`
 

>  Error: NEXUS__UNKNOWN__TYPE was already defined and imported as a type, check the docs for extending types

**Comment by Bjoernstjerne** [comment]
Fix this by edit builder.js 156 change 
"throw extendError(typeDef.name);"
to "console.log(typeDef);"
Fix your nexus API with the Errors you get now.

Three backend in production @ "nexus": "^1.3.0", with "prisma": "^4.1.1",

**Comment by guog** [comment]
- What is the status of this issue?
- Has anyone solved this problem?

**Comment by zakariamofaddel** [comment]
Having the same issue since updating to prisma v4

**Comment by Sakhro** [comment]
Same issue

**Comment by Bjoernstjerne** [comment]
Have you tried my solution to remove the throw statement in the builder? Had the same issue after upgrading from 3.1x to 4.x
I got some new errors by Nexus without the throw but this time with the type the error occured.

Holen Sie sich Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by rgamm** [comment]
I don't think this is a prisma 4 specific error. I'm getting it any time I have more than 2 missing type definitions in my nexus schema:

To reproduce, create a field on an object of `type: 'NotAType1'`, you'll get an error saying something about forgetting to export "NoteAType1"

Add a second field for 'NotAType2' and you'll get this error back instead and nothing about either NotAType1 or NotAType2:
The below code will reproduce. Comment out the second line to see a helpful error message.
```
const ObjectA = objectType({
  name: 'NotImportant',
  definition(t) {
    t.string('fine');
    t.field('errorsWhenJustThis', { type: 'NotAType1' });
    t.field('unhelpfulWithThis', { type: 'NotAType2' });
  },
});

export const schema = makeSchema({
  types: [ObjectA],
  )}
```

**Comment by guog** [comment]
please refer to here for the fix to this problem.

https://github.com/paljs/prisma-tools/issues/283#issuecomment-1287127772

**Comment by SysSU** [comment]
I'm getting this same error, I used the method [here](https://github.com/graphql-nexus/nexus/issues/1119#issuecomment-1281979445) to identify what was causing the issue, and for me one of them was the DateTime type in Prisma.

**Comment by markymc** [comment]
@SysSU I'm also getting this error with DateTime type. Did you manage to fix it?

**Comment by SysSU** [comment]
@markymc it has been some time since I looked at this code but I think the following fixed it for me.


```typescript
import { makeSchema } from 'nexus';

import {
  Json as jsonScalar,
  // Import DateTime scalar
  DateTime as dateTimeScalar,
} from 'nexus-prisma/scalars';

import * as types from './types';

export default makeSchema({
  // Add dateTimeScalar to types when calling makeSchema.
  types: { ...types, jsonScalar, dateTimeScalar },
});
```

**Comment by markymc** [comment]
@SysSU Thanks! That did it!
